File: Daily Challanges/2026-07-July/2026-07-18.js
function getOdds(dice, target) {
  let dp = new Array(target + 1).fill(0);
  dp[0] = 1;

  for (let i = 0; i < dice; i++) {
    const nextDp = new Array(target + 1).fill(0);

    for (let total = 0; total <= target; total++) {
      if (dp[total]) {
        for (let face = 1; face <= 6; face++) {
          if (total + face <= target) {
            nextDp[total + face] += dp[total];
          }
        }
      }
    }

    dp = nextDp;
  }

  const totalOutcomes = 6 ** dice;
  const ways = dp[target];
  const odds = Math.round(totalOutcomes / ways);

  return `1 in ${odds}`;
}

console.log(getOdds(1, 5));
console.log(getOdds(2, 4));
console.log(getOdds(3, 10));
console.log(getOdds(4, 7));
console.log(getOdds(5, 26));
console.log(getOdds(6, 35));

File: Daily Challanges/2026-07-July/2026-07-18.py
def get_odds(dice, target):
    dp = [0] * (target + 1)
    dp[0] = 1

    for _ in range(dice):
        next_dp = [0] * (target + 1)
        for total in range(target + 1):
            if dp[total]:
                for face in range(1, 7):
                    if total + face <= target:
                        next_dp[total + face] += dp[total]
        dp = next_dp

    total_outcomes = 6**dice
    ways = dp[target]
    odds = round(total_outcomes / ways)

    return f"1 in {odds}"


print(get_odds(1, 5))
print(get_odds(2, 4))
print(get_odds(3, 10))
print(get_odds(4, 7))
print(get_odds(5, 26))
print(get_odds(6, 35))
